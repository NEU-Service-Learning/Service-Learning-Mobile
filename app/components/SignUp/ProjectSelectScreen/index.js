import React, { Component } from 'react';
import {StyleSheet, View, Text, ActivityIndicator, TouchableHighlight } from 'react-native';

import {SearchTable, SearchRow} from '../searchTable';
import api from '../../api/index';
import style from '../../../Styles/styles';


export default class ProjectSelectScreen extends Component {

  constructor(props) {
    super(props);
    this.state = {error: false, loading: true, projects: [], selectedProjects: []}
  }

  componentWillMount() {
  try {
    let projects = [];
    let promises = [];
    this.props.classes.forEach((someClass) => {
      var projectPromise = api.getProjectForCourse(someClass.id);
      promises.push(projectPromise);
    });
    Promise.all(promises).then((data) => {
      data.forEach((classProjects) => {
        classProjects.forEach((project) => {
          console.log(project)
          projects.push(project);
        })
      })
      this.setState({loading: false, projects: projects});
    });
  } catch (e) {
    this.setState({loading: false, error: true})
    }
  }

  // Adds a project to the state when clicked
  onAddProject(addedProjct) {
    const projects = [...this.state.selectedProjects, addedProjct]
    this.setState({selectedProjects: projects});
  }

  // removes an added project from the state
  onRemoveProject(removedProject) {
    const projects = this.state.selectedProjects.filter((project) => {
       return project != removedProject
    });
    this.setState({selectedProjects: projects});
  }

  // Renders a given project's row in the table
  renderRow(add, onClickFn) {
      return  (someProj) => {
        console.log(someProj)
        return(
          <SearchRow
              onClicked={onClickFn}
              header={someProj.name}
              subHeader={someProj.description}
              data={someProj}
              key={someProj.id}
              type={add}/>
          );
      }
    }

  // For now just goes to Dashboard
  navigate() {
   this.props.navigator.push({
     title: 'Dashboard'
   });
  }

  render() {
    console.log("PR");
    console.log(this.state.projects);
    const shownProjects = this.state.projects.filter((project) => {
       return !this.state.selectedProjects.includes(project);
    });
    if(this.state.loading) {
        return(
        <View style={[style.container, style.alignCenter]}>
          <ActivityIndicator animating={true} />
        </View>
      )
    }
    return(
      <View style={styles.container}>
        <SearchTable
          style={{marginTop: 38}}
          data={this.state.projects}
          header="Avaliable Projects"
          row={this.renderRow(true, this.onAddProject.bind(this))} />
        <SearchTable
          data={this.state.selectedProjects}
          header="Added Projects"
          row={this.renderRow(false, this.onRemoveProject.bind(this))} />
        <View style={styles.nav} >
            <View style={styles.back}>
              <TouchableHighlight style={styles.button}  onPress={() => this.props.navigator.pop()}>
                <Text style={{color: 'white', fontWeight: 'bold'}}> Back </Text>
              </TouchableHighlight>
            </View>
            <View style={styles.next}>
              <TouchableHighlight style={styles.button}  onPress={() => this.navigate()}>
                <Text style={{color: 'white', fontWeight: 'bold'}}> Finish </Text>
              </TouchableHighlight>
            </View>
          </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    margin: 16,
    paddingTop: 24,
    justifyContent: 'space-between',
  },
  nav: {
    flexDirection: 'row',
    alignItems: 'flex-end',
  },
  back: {
    justifyContent:'flex-start',
    flexDirection: 'row',
    flex: 1
  },
  next: {
    justifyContent:'flex-end',
    flexDirection: 'row',
    flex: 1
  },
  button: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#ADD8E6',
    borderRadius: 64,
    width: 100,
    height: 40
  }
});
