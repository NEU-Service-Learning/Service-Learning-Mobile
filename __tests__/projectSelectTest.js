import 'react-native';
import React from 'react';
import ProjectSelectScreen from '../app/components/SignUp/ProjectSelectScreen/index';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <ProjectSelectScreen projects={['abc', 'def', 'ghi']}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
