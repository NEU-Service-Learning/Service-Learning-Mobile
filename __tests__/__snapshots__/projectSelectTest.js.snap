exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "borderRadius": 64,
      "flex": 1,
      "justifyContent": "space-between",
      "margin": 16,
      "paddingTop": 24,
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "teal",
        "borderRadius": 24,
        "flex": 1,
        "marginBottom": 10,
        "padding": 10,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      Avaliable Projects
    </Text>
    <ScrollView>
      <View
        style={
          Object {
            "backgroundColor": "skyblue",
            "borderRadius": 10,
            "flexDirection": "row",
            "height": 50,
            "marginBottom": 5,
            "marginTop": 5,
            "padding": 4,
          }
        }>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "teal",
                "flex": 2,
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }>
            abc
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "teal",
                "flex": 1,
                "fontSize": 13,
                "fontWeight": "bold",
              }
            } />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail">
            Add
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "skyblue",
            "borderRadius": 10,
            "flexDirection": "row",
            "height": 50,
            "marginBottom": 5,
            "marginTop": 5,
            "padding": 4,
          }
        }>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "teal",
                "flex": 2,
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }>
            def
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "teal",
                "flex": 1,
                "fontSize": 13,
                "fontWeight": "bold",
              }
            } />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail">
            Add
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "skyblue",
            "borderRadius": 10,
            "flexDirection": "row",
            "height": 50,
            "marginBottom": 5,
            "marginTop": 5,
            "padding": 4,
          }
        }>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "teal",
                "flex": 2,
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }>
            ghi
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "teal",
                "flex": 1,
                "fontSize": 13,
                "fontWeight": "bold",
              }
            } />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail">
            Add
          </Text>
        </View>
      </View>
    </ScrollView>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "teal",
        "borderRadius": 24,
        "flex": 1,
        "marginBottom": 10,
        "padding": 10,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      Added Projects
    </Text>
    <ScrollView />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#ADD8E6",
            "borderRadius": 64,
            "height": 40,
            "justifyContent": "center",
            "width": 100,
          },
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
          }
        }>
         Finish 
      </Text>
    </View>
  </View>
</View>
`;
